docker/Dockerfile.base

```diff
@@ -38,7 +38,9 @@ ENV SPARK_HOME /spark-2.4.0-bin-hadoop2.7
 ENV PATH "$PATH:$SPARK_HOME/sbin:$SPARK_HOME/bin"
 
 COPY docker/requirements.txt .
-RUN python3 -m pip install --no-cache-dir -U -r requirements.txt
+# re: pip install --upgrade pip: https://github.com/grpc/grpc/issues/22815
+RUN python3 -m pip install --upgrade pip keyrings.alt && \
+    python3 -m pip install --no-cache-dir --upgrade -r requirements.txt
```

Comment:
I think there's also something in the Dockerfile.services-base file and the batch/Dockerfile.worker file.

Reply:
I added the pip upgrade to Dockerfile.worker. It doesn't need to be added to services-base because that image uses base. It will already have an up-to-date version of pip.